src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -1031,13 +1048,52 @@ private async Task CommonStreaming(ConnectionState connectionState, string strea
         return await invocationTask.ConfigureAwait(false);
     }
 
+    private Activity? StartActivity(ConnectionContext connection, string methodName)
+    {
+        var connectionUri = (connection.RemoteEndPoint is UriEndPoint e) ? e.Uri : null;
+        var serviceName = connectionUri?.AbsolutePath.TrimStart('/').TrimEnd('/');
+
+        var activityName = !string.IsNullOrEmpty(serviceName)
+            ? $"{serviceName}/{methodName}"
+            : methodName;
```

Comment:
Should we check if the ActivitySource (or Critical logging) is being listened to before allocating strings?